crates/chia-consensus/src/check_time_locks.rs

```diff
@@ -0,0 +1,348 @@
+            }
+        }
+        if let Some(before_seconds_relative) = spend.before_seconds_relative {
+            if timestamp >= unspent.timestamp + before_seconds_relative {
+                return Some(ErrorCode::AssertBeforeSecondsRelativeFailed);
+            }
+        }
+    }
+
+    None
+}
+
+#[cfg(feature = "py-bindings")]
+#[pyfunction]
+#[pyo3(name = "check_time_locks")]
+pub fn py_check_time_locks(
+    removal_coin_records: &Bound<'_, PyDict>,
+    bundle_conds: &OwnedSpendBundleConditions,
+    prev_transaction_block_height: u32,
+    timestamp: u64,
+) -> PyResult<Option<u32>> {
+    let mut removals_hashmap: HashMap<chia_protocol::BytesImpl<32>, CoinRecord> = HashMap::new();
+    for (k, v) in removal_coin_records.iter() {
+        let key_bytes: Bytes32 = k.extract()?;
+        let coin_record: CoinRecord = v.extract()?;
+        removals_hashmap.insert(key_bytes, coin_record);
+    }
+
+    let res = check_time_locks(
+        &removals_hashmap,
+        bundle_conds,
+        prev_transaction_block_height,
+        timestamp,
+    );
+
+    Ok(res.map(std::convert::Into::into))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::owned_conditions::OwnedSpendConditions;
+    use chia_protocol::Coin;
+    use rstest::rstest;
+    use std::collections::HashMap;
+
+    fn dummy_coin_record(confirmed_block_index: u32, timestamp: u64) -> CoinRecord {
+        CoinRecord {
+            coin: Coin {
+                parent_coin_info: Bytes32::default(),
+                puzzle_hash: Bytes32::default(),
+                amount: 1,
+            },
+            confirmed_block_index,
+            spent_block_index: 0,
+            coinbase: false,
+            timestamp,
+        }
+    }
+
+    fn dummy_spend_conditions(
+        coin_id: Bytes32,
+        height_relative: Option<u32>,
+        seconds_relative: Option<u64>,
+        before_height_relative: Option<u32>,
+        before_seconds_relative: Option<u64>,
+        birth_height: Option<u32>,
+        birth_seconds: Option<u64>,
+    ) -> OwnedSpendConditions {
+        OwnedSpendConditions {
+            coin_id,
+            parent_id: Bytes32::default(),
+            puzzle_hash: Bytes32::default(),
+            coin_amount: 1,
+            height_relative,
+            seconds_relative,
+            before_height_relative,
+            before_seconds_relative,
+            birth_height,
+            birth_seconds,
+            create_coin: vec![],
+            agg_sig_me: vec![],
+            agg_sig_parent: vec![],
+            agg_sig_puzzle: vec![],
+            agg_sig_amount: vec![],
+            agg_sig_puzzle_amount: vec![],
+            agg_sig_parent_amount: vec![],
+            agg_sig_parent_puzzle: vec![],
+            flags: 0,
+            execution_cost: 0,
+            condition_cost: 0,
+        }
+    }
+
+    fn dummy_bundle(spends: Vec<OwnedSpendConditions>) -> OwnedSpendBundleConditions {
+        OwnedSpendBundleConditions {
```

Comment:
I think this is the same as `OwnedSpendBundleConditions { spends, ..Default::default()}`

Reply:
It would be - if `OwnedSpendBundleConditions` implemented `Default`, but it doesn't